diff --git a/dwm.c b/dwm.c
index ab3a84c..0edddac 100644
--- a/dwm.c
+++ b/dwm.c
@@ -232,6 +232,7 @@ static int xerror(Display *dpy, XErrorEvent *ee);
 static int xerrordummy(Display *dpy, XErrorEvent *ee);
 static int xerrorstart(Display *dpy, XErrorEvent *ee);
 static void zoom(const Arg *arg);
+static void redshift(const Arg *arg);
 
 /* variables */
 static const char broken[] = "broken";
@@ -2140,6 +2141,37 @@ zoom(const Arg *arg)
 	pop(c);
 }
 
+void
+redshift(const Arg *arg)
+{
+	struct sigaction sa;
+	static const char *arg_rs[] = {
+		"echo 300 > /sys/class/backlight/intel_backlight/brightness; redshift -P -O 1000K",
+		"echo 3000 > /sys/class/backlight/intel_backlight/brightness; redshift -P -O 6500K",
+		// "1000K",
+		// "6500K",
+	};
+	static int idx = 0;
+
+	if (fork() == 0) {
+		if (dpy)
+			close(ConnectionNumber(dpy));
+		setsid();
+
+		sigemptyset(&sa.sa_mask);
+		sa.sa_flags = 0;
+		sa.sa_handler = SIG_DFL;
+		sigaction(SIGCHLD, &sa, NULL);
+
+		execl("/bin/sh", "REDSHIFT", "-c", arg_rs[idx], (char *)NULL);
+		// execl("/bin/redshift", "redshift", "-P", "-O", arg_rs[idx],
+		//       (char *)NULL);
+		die("dwm: redshift failed.");
+	}
+
+	idx ^= 1;
+}
+
 int
 main(int argc, char *argv[])
 {
